docs/docs/modifying-app-ui-through-backend.md

```diff
@@ -0,0 +1,215 @@
+
+UI manager contains methods linked to each frontend component. For example, in previous code snippets we provide a `ui.Text` method, which is used for creating [Text components](https://www.streamsync.cloud/component-list.html#text).
+
+This method expects `content: dict` as first argument, which enables you to modify the field properties of the component, through corresponding keys:
+```python
+ui.Text(
+    {
+        "text": "Hello World!",  
+        # The text content of the component
+        "useMarkdown": "no",  
+        # Will not use Markdown
+        "alignment": "left",  
+        # Text is aligned to the left
+        "primaryTextColor": "#000000",  
+        # The text color is black
+        "cssClasses": "my-text hello-world"  
+        # Apply 'my-text' and 'hello-world' CSS classes
+    }
+)
+```
+
+In a similar way, every other component method also expects `content` as its first argument:
+```python
+ui.VideoPlayer(
+    {
+        "src": "https://example.com/assets/mov/rick-roll-video.mov",
+        "autoplay": "yes",
+        "controls": "no",
+        "muted": "no",
+        "loop": "no",
+    }
+)
+```
+
+In addition to `content`, a set of fields which is specific to the component type, you can also modify the base properties of the component itself, which are:
+- **`id: str`**: A unique identifier used for accessing the component after it was created.  
+    *Providing an identifier that is already taken would result in `RuntimeWarning` and the existing component being overwritten with a newly created one.*
+    ```python
+    ui.Text(
+        {"text": "Hello World!"}, 
+        id="hello-world-text"
+        )
+    ```
+    *If no ID is provided with a component, a UUID is automatically generated for it.*
+- **`position: int`**: Determines the display order of the component in relation to its siblings.  
+    Position `0` means that the component is the first child of its parent.  
+    Position `-2` is used for components – such as [sidebars](https://www.streamsync.cloud/component-list.html#sidebar) – that have a specific reserved position not related to their siblings.  
+    ```python
+    ui.Text(
+        {"text": "Hello Parent, I'm your first child!"}, 
+        position=0
+        )
+    ```
+    *Position is calculated automatically for each component, and you should be careful when you override it with predefined value, as this might lead to unexpected results.*
+- **`parentId: str`**: Determines the parent [container](#container-components) for the component. By default, components recognize the container in the context of which they were defined as their parent. This allows for linking components to their parents outside of context, or for overriding a parent within a context.
+    ```python
+    ui.Text(
+        {"text": "Hello Parent, I'm your child too!"}, 
+        parentId="dear-parent"
+        )
+    ```
+- **`visible: bool`**: Determines the visibility of the component, `True` by default.
```

Comment:
Visible should be `bool | str` / `Union[bool, str]`. Hope that's a doc oversight!

Reply:
Yeah, my bad 😅